contracts/contracts/coordination/SigningCoordinator.sol

```diff
@@ -0,0 +1,377 @@
+        SigningCohortParticipant[] signers;
+        uint256[] chains;
+        mapping(uint256 => bytes) conditions; // TODO: chainId -> condition itself or hash(condition)
+    }
+
+    bytes32 public constant INITIATOR_ROLE = keccak256("INITIATOR_ROLE");
+
+    mapping(uint32 => SigningCohort) public signingCohorts;
+    uint256 public numberOfSigningCohorts;
+
+    event InitiateSigningCohort(
+        uint32 indexed cohortId,
+        uint256 chainId,
+        address indexed authority,
+        address[] participants
+    );
+    event SigningCohortSignaturePosted(
+        uint32 indexed cohortId,
+        address indexed provider,
+        bytes signature
+    );
+    event SigningCohortDeployed(uint32 indexed cohortId, uint256 chainId);
+    event SigningCohortConditionsSet(uint32 indexed cohortId, uint256 chainId, bytes conditions);
+
+    event TimeoutChanged(uint32 oldTimeout, uint32 newTimeout);
+    event MaxCohortSizeChanged(uint16 oldSize, uint16 newSize);
+    event DispatcherChanged(address oldDispatcher, address newDispatcher);
+
+    enum SigningCohortState {
+        NON_INITIATED,
+        AWAITING_SIGNATURES,
+        TIMEOUT,
+        ACTIVE,
+        EXPIRED
+    }
+
+    TACoApplication public immutable application;
+    SigningCoordinatorDispatcher public signingCoordinatorDispatcher;
+
+    uint96 private immutable minAuthorization;
+
+    uint32 public timeout;
+    uint16 public maxCohortSize;
+
+    SigningCohortParticipant internal __sentinelSigner;
+
+    constructor(TACoApplication _application) {
+        application = _application;
+        minAuthorization = _application.minimumAuthorization();
+        _disableInitializers();
+    }
+
+    function initialize(
+        uint32 _timeout,
+        uint16 _maxDkgSize,
+        SigningCoordinatorDispatcher _signingCoordinatorDispatcher,
+        address _admin
+    ) external initializer {
+        timeout = _timeout;
+        maxCohortSize = _maxDkgSize;
+        signingCoordinatorDispatcher = _signingCoordinatorDispatcher;
+        __AccessControlDefaultAdminRules_init(0, _admin);
+    }
+
+    function setTimeout(uint32 newTimeout) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        emit TimeoutChanged(timeout, newTimeout);
+        timeout = newTimeout;
+    }
+
+    function setMaxDkgSize(uint16 newSize) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        emit MaxCohortSizeChanged(maxCohortSize, newSize);
+        maxCohortSize = newSize;
+    }
+
+    // TODO: this should be removed post-testing
+    function setDispatcher(
+        SigningCoordinatorDispatcher dispatcher
+    ) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        require(address(dispatcher).code.length > 0, "Dispatcher must be contract");
+        emit DispatcherChanged(address(signingCoordinatorDispatcher), address(dispatcher));
+        signingCoordinatorDispatcher = dispatcher;
+    }
+
+    function _isCohortActive(SigningCohort storage _cohort) internal view returns (bool) {
```

Comment:
no need in underscore because function has different parameters compering to other `isCohortActive`
same about `_cohort` - there is no other variable with the same name so no need to cover it with underscore

Reply:
I added the underscore to the function name because it is an internal function. Is that the convention or was I mistaken?

Reply:
yeah, it is underscore only for cases when there is function/variable with that name.

Reply:
✔️